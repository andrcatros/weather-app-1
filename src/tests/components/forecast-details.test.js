import React from 'react'
import { render } from '@testing-library/react'
import ForecastDetails from '../../components/forecastdetails'

describe('Forecast Details', () => {
    const forecast = {
        date: 1525046400000,
        temperature: {
            max: 11,
            min: 4
        },
        wind: {
            speed: 13,
            direction: "s"
        },
        humidity: 30,
        description: "Clear",
        icon: "800"
    }

    it('renders correctly', () => {
        const { asFragment } = render(
            <ForecastDetails
                forecast={forecast}
            />
        )

        expect(asFragment()).toMatchSnapshot();
    })

    it('renders a number of props into a weather summary', () => {
        const { getByText, getByTestId } = render(
            <ForecastDetails
                forecast={forecast}
            />
        )

        expect(getByText('April 30th, 2018')).toBeTruthy();
        expect(getByText('April 30th, 2018')).toHaveClass('date');
        expect(getByTestId('max-temperature-id')).toBeInTheDocument();
        expect(getByTestId('min-temperature-id')).toBeInTheDocument();
        expect(getByText('Humidity: 30%')).toHaveClass('humidity');
        expect(getByText('Wind: 13mph')).toHaveClass('wind');
    })
})