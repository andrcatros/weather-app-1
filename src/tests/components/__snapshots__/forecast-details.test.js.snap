// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast Details renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <h2
      class="date"
      data-testid="date-id"
    >
      April 30th, 2018
    </h2>
    <span
      class="max-temperature"
      data-testid="max-temperature-id"
    >
      Max Temperature: 11℃
    </span>
    <span
      class="min-temperature"
      data-testid="min-temperature-id"
    >
      Min Temperature: 4℃
    </span>
    <span
      class="humidity"
      data-testid="humidity-id"
    >
      Humidity: 30%
    </span>
    <span
      class="wind"
      data-testid="wind-id"
    >
      Wind: 13mph
    </span>
  </div>
</DocumentFragment>
`;
